File: CSharp/DeepNestedComplexity.cs
namespace Codescene.VSExtension.CodeSmells.Issues.CSharp
{
    class DeepNestedComplexity
    {
        public bool IsDead { get; set; }
        public bool IsSeparated { get; set; }
        public bool IsRetired { get; set; }

        private decimal DeadAmount() => 0m;
        private decimal SeparatedAmount() => 100m;
        private decimal RetiredAmount() => 200m;
        private decimal NormalPayAmount() => 300m;

        public decimal GetPayAmount()
        {
            decimal result;
            if (IsDead)
            {
                result = DeadAmount();
            }
            else
            {
                if (IsSeparated)
                {
                    result = SeparatedAmount();
                }
                else
                {
                    if (IsRetired)
                    {
                        result = RetiredAmount();
                    }
                    else
                    {
                        result = NormalPayAmount();
                    }
                }
            }
            return result;
        }
    }

}

File: Javascript/DeepNestedComplexity.js
class DeepNestedComplexity {
    constructor() {
        // Initialize properties. You can change these as needed.
        this.isDead = false;
        this.isSeparated = false;
        this.isRetired = false;
    }

    // Private methods (by convention, using underscore)
    _deadAmount() {
        return 0;
    }

    _separatedAmount() {
        return 100;
    }

    _retiredAmount() {
        return 200;
    }

    _normalPayAmount() {
        return 300;
    }

    // Public method: gets the pay amount based on the nested conditionals.
    getPayAmount() {
        let result;
        if (this.isDead) {
            result = this._deadAmount();
        } else {
            if (this.isSeparated) {
                result = this._separatedAmount();
            } else {
                if (this.isRetired) {
                    result = this._retiredAmount();
                } else {
                    result = this._normalPayAmount();
                }
            }
        }
        return result;
    }
}

// Example usage:
const example = new DeepNestedComplexity();

// Set conditions
example.isDead = false;
example.isSeparated = true;
example.isRetired = false;

console.log(example.getPayAmount()); // Output: 100, since isSeparated is true
